Fix Play Store uploads in CI

I need to make a new Android upload keystore and add it to GitHub Actions. Here's the failed workflow job, "Submit Android build": https://github.com/lockdown-systems/cyd-mobile/actions/runs/22510391857/job/65219598531

```
Run node ./scripts/ci/submit-android.js dist/android/cyd-mobile.aab
Uploading cyd-mobile.aab (102.3 MB) to Google Play Console...
  Package: systems.lockdown.cydmobile
  Track:   internal
  Status:  completed

Creating edit...
Uploading bundle (this may take a moment)...

Google Play API error:
  403 The Android App Bundle was signed with the wrong key. Found: SHA1: 5E:8F:16:06:2E:A3:CD:2C:4A:0D:54:78:76:BA:A6:F3:8C:AB:F6:25, expected: SHA1: 97:AF:BC:5D:80:9F:0A:E3:CE:CF:59:74:9D:B4:58:B3:55:7E:49:03.
  - The Android App Bundle was signed with the wrong key. Found: SHA1: 5E:8F:16:06:2E:A3:CD:2C:4A:0D:54:78:76:BA:A6:F3:8C:AB:F6:25, expected: SHA1: 97:AF:BC:5D:80:9F:0A:E3:CE:CF:59:74:9D:B4:58:B3:55:7E:49:03. (forbidden)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -184,6 +184,11 @@ jobs:
     name: Build Android
     runs-on: ubuntu-latest-android-builder
     needs: quality
+    env:
+      ANDROID_KEYSTORE_BASE64: ${{ secrets.ANDROID_KEYSTORE_BASE64 }}
+      ANDROID_KEYSTORE_PASSWORD: ${{ secrets.ANDROID_KEYSTORE_PASSWORD }}
+      ANDROID_KEY_ALIAS: ${{ secrets.ANDROID_KEY_ALIAS }}
+      ANDROID_KEY_PASSWORD: ${{ secrets.ANDROID_KEY_PASSWORD }}
 
     steps:
       - name: Checkout repository
@@ -210,8 +215,26 @@ jobs:
       - name: Generate native iOS/Android projects
         run: npm run prebuild
 
+      - name: Decode upload keystore
+        run: |
+          echo "$ANDROID_KEYSTORE_BASE64" | base64 --decode > android/app/upload.jks
+
+      - name: Configure release signing
+        run: |
+          python3 ./scripts/ci/configure-android-signing.py
+          echo "--- Signing config in build.gradle ---"
+          grep -A5 'signingConfigs' android/app/build.gradle | head -25
+
       - name: Tune Gradle for CI runner
         run: |
+          cat >> android/gradle.properties <<EOF
+
+          # Upload keystore credentials
+          UPLOAD_STORE_PASSWORD=$ANDROID_KEYSTORE_PASSWORD
+          UPLOAD_KEY_ALIAS=$ANDROID_KEY_ALIAS
+          UPLOAD_KEY_PASSWORD=$ANDROID_KEY_PASSWORD
+          EOF
+
           cat >> android/gradle.properties <<'EOF'
 
           # CI overrides for 8-core / 32GB runner

--- a/scripts/ci/configure-android-signing.py
+++ b/scripts/ci/configure-android-signing.py
@@ -1,0 +1,33 @@
+#!/usr/bin/env python3
+"""Patch the Expo-generated build.gradle to use the upload keystore for release signing."""
+
+path = "android/app/build.gradle"
+with open(path) as f:
+    content = f.read()
+
+# Add release signing config after the debug block
+release_config = """
+        release {
+            storeFile file("upload.jks")
+            storePassword findProperty("UPLOAD_STORE_PASSWORD") ?: ""
+            keyAlias findProperty("UPLOAD_KEY_ALIAS") ?: ""
+            keyPassword findProperty("UPLOAD_KEY_PASSWORD") ?: ""
+        }"""
+
+content = content.replace(
+    "storePassword 'android'\n            keyAlias 'androiddebugkey'\n            keyPassword 'android'\n        }\n    }",
+    "storePassword 'android'\n            keyAlias 'androiddebugkey'\n            keyPassword 'android'\n        }"
+    + release_config
+    + "\n    }",
+)
+
+# Use release signing config for release builds
+content = content.replace(
+    "release {\n            // Caution! In production, you need to generate your own keystore file.\n            // see https://reactnative.dev/docs/signed-apk-android.\n            signingConfig signingConfigs.debug",
+    "release {\n            signingConfig signingConfigs.release",
+)
+
+with open(path, "w") as f:
+    f.write(content)
+
+print("Patched build.gradle for release signing")

